client/my-sites/checkout/composite-checkout/use-get-thank-you-url.js

```diff
@@ -188,13 +189,26 @@ function getFallbackDestination( {
 	// - has a receipt number
 	// - does not have a receipt number but has an item in cart(as in the case of paying with a redirect payment type)
 	if ( siteSlug && ( ! isReceiptEmpty || ! isCartEmpty ) ) {
-		const isJetpackProduct = product && JETPACK_BACKUP_PRODUCTS.includes( product );
-		// If we just purchased a Jetpack product, redirect to the my plans page.
-		if ( isJetpackNotAtomic && isJetpackProduct ) {
-			debug( 'the site is jetpack and bought a jetpack product', siteSlug, product );
-			return `/plans/my-plan/${ siteSlug }?thank-you=true&product=${ product }`;
+		// If we just purchased a Jetpack product or a Jetpack plan (either Jetpack Security or Jetpack Complete),
+		// redirect to the my plans page. The product being purchased can come from the `product` prop or from the
+		// cart so we need to check both places.
+		const productsWithCustomThankYou = [ ...JETPACK_PRODUCTS_LIST, ...JETPACK_RESET_PLANS ];
+
+		// Check the cart (since our Thank You modal doesn't support multiple products, we only take the first
+		// one found).
+		const productFromCart = cart?.products?.find( ( { product_slug } ) =>
+			productsWithCustomThankYou.includes( product_slug )
+		)?.product_slug;
+
+		const purchasedProduct =
+			productFromCart ||
+			productsWithCustomThankYou.find( ( productWithCustom ) => productWithCustom === product );
```

Comment:
This line broke the flow in the previous PR because I wrote `productsWithCustomThankYou.find( product );` which obviously was not going to work since `find` requires a function as parameter.